### PR TITLE
Clean up session callbacks properly

### PR DESCRIPTION
**Pull request by silverbucket**
This is a bugfix to remove callbacks when a session is terminated.

### CODE REVIEW

File: packages/sockethub/src/janitor.ts [hunk 1]
```diff
@@ -57,6 +57,9 @@ function removeStaleSocketSessions(platformInstance, sockets) {
       rmLog('removing stale socket session reference ' + sessionId + ' in platform instance '
         + platformInstance.id);
       platformInstance.sessions.delete(sessionId);
+      for (let key in platformInstance.sessionCallbacks) {
+        platformInstance.sessionCallbacks[key].delete(sessionId);
+      }
```
Comment (raucao) [inline packages/sockethub/src/janitor.ts]: I think in modern JS, you could write this like so:

```js
      for (const sessionCallback of platformInstance.sessionCallbacks) {
        sessionCallback.delete(sessionId);
      }
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of

Comment (silverbucket) [reply packages/sockethub/src/janitor.ts]: This isn't the case for objects, you can't iterate over the values of them without some additional code.

Comment (raucao) [reply packages/sockethub/src/janitor.ts]: Ah, sorry. I thought it was an array for some reason.

**Comment by silverbucket** [comment]
@raucao think I could get an approval on this one?

**Pull request merged**